akka-stream/src/main/scala/akka/stream/impl/io/InputStreamPublisher.scala

```diff
@@ -87,8 +87,6 @@ private[akka] class InputStreamPublisher(is: InputStream, bytesReadPromise: Prom
 
   /** BLOCKING I/O READ */
   def loadChunk() = try {
-    val arr = Array.ofDim[Byte](chunkSize)
```

Comment:
@drewhk The difference is that we'd retain the shared array between reads with 0 additional benefit (besides not using an internal API)